spec/defines/nexus_spec.rb

```diff
@@ -0,0 +1,68 @@
+require 'spec_helper'
+
+describe 'archive::nexus' do
+  context 'nexus archive with defaults' do
+    let :title do
+      '/tmp/hawtio.war'
+    end
+
+    let :params do
+      {
+        url: 'https://oss.sonatype.org',
+        gav: 'io.hawt:hawtio-web:1.4.36',
+        repository: 'releases',
+        packaging: 'war'
+      }
+    end
+
+    it do
+      should contain_archive('/tmp/hawtio.war').with(
+        'source' => 'https://oss.sonatype.org/service/local/artifact/maven/content?g=io.hawt&a=hawtio-web&v=1.4.36&r=releases&p=war',
+        'checksum_url' => 'https://oss.sonatype.org/service/local/artifact/maven/content?g=io.hawt&a=hawtio-web&v=1.4.36&r=releases&p=war.md5'
+      )
+    end
+
+    it do
+      should contain_file('/tmp/hawtio.war').that_requires('Archive[/tmp/hawtio.war]').with(
+        'owner' => '0',
+        'group' => '0'
+      )
+    end
+  end
+
+  context 'nexus archive with overwritten parameters' do
+    let :title do
+      '/tmp/artifact.war'
+    end
+
+    let :params do
+      {
+        owner: 'tom',
+        group: 'worker',
+        user: 'tom',
+        extract: true,
+        extract_path: '/opt',
+        creates: '/opt/artifact/WEB-INF',
+        cleanup: true
+      }
```

Comment:
gav and repository are required parameters, and packaging, when supressed, defaults to 'jar' (in Nexus API), so I made packaging defaults to 'jar' in order to produce the same effect.

Still getting the error. I'll investigate with pry.